suite-native/navigation/src/hooks/useOverrideBackNavigation.tsx

```diff
@@ -2,7 +2,9 @@ import { useEffect } from 'react';
 
 import { useNavigation } from '@react-navigation/native';
 
-export const useOverrideBackNavigation = ({ onNavigateBack }: { onNavigateBack?: () => void }) => {
+export const useOverrideBackNavigation = ({
+    onNavigateBack,
+}: { onNavigateBack?: () => void } = {}) => {
```

Comment:
Actually I'm thinking whether it would make sense to use route matching for this and if the screen that it is intended for is focused. Since some of our screens don't unmount, it would be nice that this would only work on the actual screen where this hook is initialized?